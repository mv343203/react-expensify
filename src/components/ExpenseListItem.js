import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


//destructure props object to access to individual data  from the props object 
//we will then import into the ExpenseList 
//access the store by adding dispatch into the spread and id to grab specefic id 
const ExpenseListItem = ({id, description, amount, createdAt}) => (
     <Link className="list-item" to = {`/edit/${id}`}> 
      <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__sub-title"> {moment(createdAt).format('MMMM Do, YYYY')} </span>
      </div> 
      <h3 className='list-item__data'>{numeral(amount / 100).format ('$0,0.00')} </h3>
    </Link>
);

//default 
export default ExpenseListItem;