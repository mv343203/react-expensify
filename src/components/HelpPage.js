import React from 'react';


const HelpPage  = () => (
    <div>
    HELPPPP!!
    </div>
);

export default HelpPage;