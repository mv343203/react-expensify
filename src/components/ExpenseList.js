import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'


//map <MAP DESCRIPTION>the expenses array which allows you to take in an array and get back something else 
    //requires you to set an updater function 
    //function takes in indvidual expense 
    //Then returns its JSX and we use spread operator to spread out its properties 
    // a key prop has to be set too to avoid error and has to be a uniqe value (which is key value
// in our case we are getting in an array of objects and returning list items 

export const ExpenseList = (props) => (
  <div className = 'content-container'>
    <div className='list-header'>
      <div className='show-for-mobile'>Expenses</div>
      <div className='show-for-desktop'>Expense</div>
      <div className='show-for-desktop'>Amount</div>
    </div>
    <div className="list-body"> 
    {
      props.expenses.length === 0 ? (
       <div className="list-item list-item--message"> 
        <span> No Expenses</span>
       </div>
      ) : (

        props.expenses.map((expense) => {
          return <ExpenseListItem key ={expense.id} {...expense} />;
        })
      )
      }
      </div>
  </div>
);

//imported selectExpenses to sort the data
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
   };

};

//pulls everything together 
export default connect (mapStateToProps)(ExpenseList);

