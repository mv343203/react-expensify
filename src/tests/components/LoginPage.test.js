import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';



test ('should correctly render LoginPage', () =>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    //creates spy
    const startLogin = jest.fn();
    //passes in to login page
    const wrapper = shallow (<LoginPage startLogin = {startLogin} />);
    //this clicks the button
    wrapper.find('button').simulate('click');
    //expects spy to have been called
    expect(startLogin).toHaveBeenCalled();
});