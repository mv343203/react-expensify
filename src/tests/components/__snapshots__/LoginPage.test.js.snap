// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layoyt__title"
    >
       Expensify
    </h1>
    <p>
       Take Control Of Your Expenses
    </p>
    <button
      className="button"
    >
       Login With Google 
    </button>
  </div>
</div>
`;
